### DISCUSSION TITLE
Header blur effect on scroll

### DISCUSSION THREAD

**Discussion by nandorojo** [software-mansion/react-native-screens#1229] [Q&A]
GitHub's iOS app doesn't show the header blur until you've scrolled down. Is there a way to achieve this? Perhaps with animating in the blur as you scroll?

https://user-images.githubusercontent.com/13172299/144724362-45fbe19e-7817-4d40-9911-2207c4ecf17d.MOV

**Comment by WoLewicki** [comment]
Cannot you use `navigation.setOptions` with `blurEffect` based on the scroll position?

**Reply by nandorojo** [reply]
Yeah, although it wouldn’t animate the blur right?

**Reply by WoLewicki** [reply]
No, it won't, but so it doesn't in Github app right? It wouldn't be easy to make it animatable, have you seen any apps where the blur effect is animated?

**Reply by nandorojo** [reply]
I think I saw one recently but forgot to get a video. I’ll try to find it. Alternatively, I could render a custom BlurView there somehow, maybe with header transparent.

**Reply by WoLewicki** [reply]
Custom `BlurView` may be a way to go for sure since the native solution may be not achievable easily.

**Reply by nandorojo** [reply]
https://user-images.githubusercontent.com/13172299/148313964-3e57c621-973d-4b8b-a60a-0faa850961f8.MOV

iMessage has the faded blur header.

**Reply by nandorojo** [reply]
similarly, is there a way to add headerBottomContent like these pills on the GitHub notifications that are sticky under the title?

https://user-images.githubusercontent.com/13172299/148315884-5428519f-65fc-44bc-b133-410ac9e931e6.MOV

**Reply by WoLewicki** [reply]
> Upload.from.GitHub.for.iOS.MOV 
> iMessage has the faded blur header.

Can't you achieve something like this with something like such configuration of header: 
```tsx
screenOptions={{
        headerTranslucent: true,
        headerStyle: {
          backgroundColor: 'transparent',
          blurEffect: 'systemUltraThinMaterialDark',
        },
        headerLargeStyle: {
          backgroundColor: 'black',
        },
        headerLargeTitle: true,
}}
```

and dark theme on the `NavigationContainer` ?


> similarly, is there a way to add headerBottomContent like these pills on the GitHub notifications that are sticky under the title?
> 
>  Upload.from.GitHub.for.iOS.MOV

I think it is achieved in native apps just by sticky header in `ScrollView`. Is it not possible to do something like this in your project too?

**Reply by cnhhoang850** [reply]
This is the way

**Reply by EvanBacon** [reply]
@WoLewicki this no longer works in SDK 52 / React Navigation v7 / `react-native-screens@4.0.0`. The `headerLargeStyle.backgroundColor` is not respected and the header stays blurry.

**Reply by EvanBacon** [reply]
These two changes get closer:
- https://github.com/software-mansion/react-native-screens/pull/2500
- https://github.com/software-mansion/react-native-screens/pull/2501

Main thing I'm not sure about is which blurStyle to use to match the default.

**Comment by Karthik-B-06** [comment]
@nandorojo I tried to achieve a blur effect on the scroll. It was unable to upload a video, file size was high.  Can you check the video [here](https://twitter.com/timelessco/status/1700196470748790836), and tell me if it is close to what you wanted to achieve?
